Tests/RegexTests/MatchTests.swift

```diff
@@ -30,7 +44,8 @@ func _firstMatch(
 ) throws -> (String, [String?])? {
   var regex = try Regex(regexStr, syntax: syntax).matchingSemantics(semanticLevel)
   let result = try regex.firstMatch(in: input)
-
+  try _roundTripLiteral(regexStr, syntax: syntax)
```

Comment:
Long term, we'll want to sink this deeper into the library. 

For this, I think we'd also want to test parity for parse only tests.